src/iocore/net/SSLStats.cc

```diff
@@ -328,9 +329,37 @@ SSLInitializeStatistics()
   for (const char *name : group_list) {
     add_group_stat<std::string>(name, name);
   }
-#elif defined(SSL_get_negotiated_group)
+#elif HAVE_SSL_CTX_GET0_IMPLEMENTED_GROUPS
+  // Use SSL_CTX_get0_implemented_groups to discover all supported groups.
+  // This handles KEMs which don't have standard NIDs defined.
+  {
+    SSLMultiCertConfigLoader loader(nullptr);
+    SSL_CTX                 *ctx = loader.default_server_ssl_ctx();
+    SSL                     *ssl = SSL_new(ctx);
+
+    // Get the list of implemented groups (all=1 for all implemented groups)
+    const int *groups     = nullptr;
+    int        num_groups = SSL_CTX_ctrl(ctx, 139 /* SSL_CTRL_GET0_IMPLEMENTED_GROUPS */, 1, (void *)&groups);
```

Comment:
FWIW, you can find example of `SSL_CTX_get0_implemented_groups` usage in the source of `openssl list` command.

https://github.com/openssl/openssl/blob/c8b4a397d066857492c714735d7658cd1e545e81/apps/list.c#L865

Reply:
Thank you. That's good confirmation. I independently implemented that before seeing this (see the updated patch - it's pretty identical). Sadly, the call simply hangs at the time that it is called. I'll try to see whether I can find a later stage in the code where it can be called.